modules/core/src/v2/baseCoin.ts

```diff
@@ -347,6 +348,14 @@ export abstract class BaseCoin {
     return signMessage(message, bip32.fromBase58(key.prv), utxolib.networks.bitcoin);
   }
 
+  /**
+   * Decompose a raw transaction into useful information.
+   * @param options - coin-specific
+   */
+  explainTransaction(options: Record<string, any>): Promise<TransactionExplanation<any, string | number> | undefined> {
```

Comment:
Is it possible to make this `abstract` to force inheritors to implement? Or is that going to drag in a lot more work?

I'd rather have the concrete implementations throw `not implemented` if they so choose instead of making it the default behavior in the base class.

Reply:
interestingly, a lot of coins do not implement this method at all

```
src/v2/coins/erc20Token.ts(22,14): error TS2515: Non-abstract class 'Erc20Token' does not implement inherited abstract member 'explainTransaction' from class 'Eth'.
src/v2/coins/eth.ts(334,14): error TS2515: Non-abstract class 'Eth' does not implement inherited abstract member 'explainTransaction' from class 'BaseCoin'.
src/v2/coins/eth2.ts(107,14): error TS2515: Non-abstract class 'Eth2' does not implement inherited abstract member 'explainTransaction' from class 'BaseCoin'.
src/v2/coins/gteth.ts(6,14): error TS2515: Non-abstract class 'Gteth' does not implement inherited abstract member 'explainTransaction' from class 'Eth'.
src/v2/coins/ofc.ts(19,14): error TS2515: Non-abstract class 'Ofc' does not implement inherited abstract member 'explainTransaction' from class 'BaseCoin'.
src/v2/coins/ofcToken.ts(29,14): error TS2515: Non-abstract class 'OfcToken' does not implement inherited abstract member 'explainTransaction' from class 'Ofc'.
src/v2/coins/susd.ts(17,14): error TS2515: Non-abstract class 'Susd' does not implement inherited abstract member 'explainTransaction' from class 'BaseCoin'.
src/v2/coins/teth.ts(6,14): error TS2515: Non-abstract class 'Teth' does not implement inherited abstract member 'explainTransaction' from class 'Eth'.
src/v2/coins/teth2.ts(5,14): error TS2515: Non-abstract class 'Teth2' does not implement inherited abstract member 'explainTransaction' from class 'Eth2'.
src/v2/coins/tsusd.ts(8,14): error TS2515: Non-abstract class 'Tsusd' does not implement inherited abstract member 'explainTransaction' from class 'Susd'.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```